### PR TITLE
Feature/sdk 1171

### PR DESCRIPTION
**Pull request by Guitouille**

### CODE REVIEW

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/core/JourneysRequest.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 import org.joda.time.DateTime;
 
+import org.joda.time.DateTime;
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/core/JourneysRequest.java]: Duplicate import

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/data/PartnersTicketRepository.java [hunk 2]
```diff
@@ -37,9 +48,57 @@ public PartnersTicketRepository(Gson gson) {
                                 new TypeToken<List<HermaasTicketDomain>>() {
                                 }.getType());
                         hermaasResponseDomain.setHermaasTicketDomainList(hermaasTicketDomainList);
+                        setNavitiaTicketPriceInHermaasTicket(hermaasTicketDomainList, ticketModelList);
+                    }
+
+                    String result = null;
+                    if(hermaasResponseDomain.getHermaasTicketDomainList() != null) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/data/PartnersTicketRepository.java]: Add space after if

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/data/mapper/RoadmapDomainDataMapper.java [hunk 3]
```diff
@@ -78,9 +81,12 @@ public RoadmapDomain transform(Journey journey,
             roadmapDomain.setNoteDomainList(noteDomainDataMapper.transform(noteList));
         }
 
+/*
         if (ticketList != null) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/data/mapper/RoadmapDomainDataMapper.java]: Clean code or add comment/todo

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/data/mapper/RoadmapDomainDataMapper.java [hunk 4]
```diff
@@ -102,6 +108,7 @@ private int prepareTravelDuration(Durations durations) {
         return journeySections;
     }
 
+/*
     private float getTotalPrice(List<Ticket> ticketList) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/data/mapper/RoadmapDomainDataMapper.java]: Clean code or add comment/todo

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java [hunk 5]
```diff
@@ -8,86 +8,129 @@
...
+            if (JourneysUI.getInstance().isWithBooking() && hermaasTicketDomainList.size() > 0) {
+                SectionDomain emptySectionDomain = new SectionDomain();
+                emptySectionDomain.setTypeView(VIEW_TYPE_EMPTY);
+
+                roadmapDomain.getSectionDomainList().add(emptySectionDomain);
+            }
+
+            addTaxiTimeInfoSection(roadmapDomain, infoSection);
+
+            if (infoSection.size() > 0) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java]: We don't need to check 'infoSection' size before foreach.

Comment (nicolas-chaix) [reply journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java]: Just make sure it's not null

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java [hunk 6]
```diff
@@ -101,6 +144,8 @@ private void setTypeViewOfSection(RoadmapDomain roadmapDomain) {
                     Section.ModeEnum mode = sectionDomain.getMode();
                     if (mode == WALKING || mode == CAR || mode == BIKE) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java]: A 'switch' may be cleaner

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java [hunk 7]
```diff
@@ -342,6 +488,24 @@ private int getWaitingDuration(List<SectionDomain> sectionDomainList, int index)
...
     }
 
+    private float findTicket(Section.ModeEnum mode, String ticketId, List<HermaasTicketDomain> hermaasTicketDomainList) {
+        for (HermaasTicketDomain hermaasTicketDomain : hermaasTicketDomainList) {
+            if (hermaasTicketDomain.getTicketId().equalsIgnoreCase(ticketId)) {
+                return hermaasTicketDomain.getPriceWithTax();
+            }
+
+            if (ticketId == null && mode == TAXI && String.valueOf(JourneysUI.getInstance().getIdProducts().get(mode.getValue())).equals(hermaasTicketDomain.getIdProduct())) {
+                if (hermaasTicketDomain.getStatus().equalsIgnoreCase("200")) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java]: equalsIgnoreCase is unnecessary: use equals

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/model/PublicTransportSectionRoadmapModel.java [hunk 8]
```diff
@@ -12,7 +12,7 @@
 
 public class PublicTransportSectionRoadmapModel extends SectionRoadmapModel {
     private SectionMode mode;
-    private String take;
+    private SpannableStringBuilder take;
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/model/PublicTransportSectionRoadmapModel.java]: Add TODO for changing 'SpannableStringBuilder' into 'SpannableString'

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/PublicTransportStepViewHolder.java [hunk 9]
```diff
@@ -121,8 +135,7 @@ public void fillView(PublicTransportSectionRoadmapModel publicTransportSectionRo
             gNetwork.setVisibility(VISIBLE);
         }
 
-        gTransit.setText(publicTransportSectionRoadmapModel.getTransit());
-
+/*
         if (!TextUtils.isEmpty(publicTransportSectionRoadmapModel.getLineCode())) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/PublicTransportStepViewHolder.java]: Please explain why this code is disabled

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/TaxiStepViewHolder.java [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        gTake.setText(taxiSectionRoadmapModel.getTake());
+
+        gTransitOrigin.setText(taxiSectionRoadmapModel.getOrigin());
+        gTransitStartTime.setText(taxiSectionRoadmapModel.getDepartureTime());
+        gTransitDestination.setText(taxiSectionRoadmapModel.getDestination());
+        gTransitEndTime.setText(taxiSectionRoadmapModel.getArrivalTime());
+
+
+/*
+        if (JourneysUI.getInstance().isWithMaas()) {
```
Comment (nicolas-chaix) [inline journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/TaxiStepViewHolder.java]: explain this

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java (hunk 5 above)
Comment (Guitouille) [reply journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java]: We do not need to check if it is null as it is initialized with an empty arrayList.

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java (hunk 6 above)
Comment (Guitouille) [reply journey/src/main/java/com/kisio/navitia/sdk/ui/journey/domain/interactor/GetRoadmap.java]: I think nested switch will be harder to read.

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/model/PublicTransportSectionRoadmapModel.java (hunk 8 above)
Comment (Guitouille) [reply journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/model/PublicTransportSectionRoadmapModel.java]: No need to change as the SpannableString is immutable and the SpannableStringBuilder is mutable.

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/PublicTransportStepViewHolder.java (hunk 9 above)
Comment (Guitouille) [reply journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/PublicTransportStepViewHolder.java]: I have added an TODO

File: journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/TaxiStepViewHolder.java (hunk 10 above)
Comment (Guitouille) [reply journey/src/main/java/com/kisio/navitia/sdk/ui/journey/presentation/roadmap/viewholder/TaxiStepViewHolder.java]: I have added an TODO

**Pull request merged**